config/tests/jobs/jobs_test.go

```diff
@@ -303,6 +304,40 @@ func findRequired(t *testing.T, presubmits []cfg.Presubmit) []string {
 	return required
 }
 
+func TestTrustedJobs(t *testing.T) {
+	// TODO(fejta): allow each config/jobs/kubernetes/foo/foo-trusted.yaml
+	// that uses a foo-trusted cluster
+	const trusted = "test-infra-trusted"
+	trustedPath := path.Join(*jobConfigPath, "kubernetes", "test-infra", "test-infra-trusted.yaml")
```

Comment:
I do not think basename is the right way to go here.

I don't see how portability is a concern here. This test uses `*jobConfigPath` to load the config:
https://github.com/kubernetes/test-infra/blob/fe4a6c02edc8bf39c8ee83b55a968ed0c9a75a40/config/tests/jobs/jobs_test.go#L58

So everything it finds with be relative to `jobConfigPath`. The specific value of `jobConfigPath` is irrelevant to this test. I could validate that the path ends with `/kubernetes/test-infra/test-infra-trusted.yaml` but is less correct (I can have whatever I want in front) and not any easier to read.

Just checking the basename would allow me to move this to `config/jobs/fejternetes/test-infra-trusted.yaml` which is not what we want. These are jobs trusted by the test-infra team (as it gets access to important test-infra credentials)

Also prow's config library will happily load multiple config files with the same basename. We have this restriction because a) we send these files to prow by adding them to a configmap and b) configmaps only support a flag namespace.

If we were to use a PD we could have a more detailed hierarchy with non-unique basenames.

Reply:
When we load it from the configmap it will only have basename effectively. We would actually have to use a PD and a more complex heirarchy, but currently it will not and the path will not match the one in the repo so the presubmit and deployed trusted paths will have to be different which seems not ideal.

Reply:
We could always change it when we change how we manage config, but right now it seems error prone to have a unit test trusted path and a deployed in cluster trusted path. The in cluster one only has a meaningful value for the filename currently.

Reply:
There's only a unit test trusted path.

Reply:
(discussed more offline, I agree and am on board) in the future we may revisit the entire config updater mechanism.